shared/app/nav.js.flow

```diff
@@ -1,24 +1,28 @@
 // @flow
-import {Component} from 'react'
-import {Map} from 'immutable'
 import * as React from 'react'
 
-import type {RouteRenderStack} from '../route-tree/render-route'
+import type {RouteProps} from '../route-tree/render-route'
 import type {Reachable} from '../constants/types/flow-types'
 import type {Tab} from '../constants/tabs'
 
-export type Props = {
+export type StateProps = {
+  _me: ?string,
+  appFocused: boolean,
+  reachable: Reachable,
+  hideNav?: boolean,
+}
+
+export type DispatchProps = {
+  _switchTab: (Tab, ?string) => void,
+}
+
+export type OwnProps = RouteProps<{}, {}>
+
+export type Props = OwnProps & {
```

Comment:
@chrisnojima it looks like the types here haven't been updated in a while. These props reflect what I see in the debuggers.